app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 class PostsController < ApplicationController
+  before_action :require_login, only: [:new, :create]
```

Comment:
You could also use   `  before_action :authenticate_user!` which is generated by devise